packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -826,6 +826,7 @@ class _DropdownMenuDefaultsM3 extends DropdownMenuThemeData {
     return const MenuStyle(
       minimumSize: MaterialStatePropertyAll<Size>(Size(_kMinimumWidth, 0.0)),
       maximumSize: MaterialStatePropertyAll<Size>(Size.infinite),
+      visualDensity: VisualDensity.standard,
```

Comment:
@QuncCccccc this seems partially a bit unrelated to the issue. It will make the menu unresponsive the general theme `VisualDensity` changes. Unless the used `TextField` in the `DropdownMenu` already is that as well already. Then it makes sense to make sure sizes always line up. Are all the other menus also locked and not changing with `VisualDensity`? If so (I will check) it makes even more sense. If we can avoid inconsistencies that is always nice. I have heard feedback from Flutter devs that were raging about inconsistencies in `VisualDenisity` behavior in general in Flutter, but that is another topic.

Nice find with the `key`, would have taken me a while to figure that out. Thanks for the quick fix.